src/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_img2img.py

```diff
@@ -773,6 +819,9 @@ def __call__(
         aesthetic_score: float = 6.0,
         negative_aesthetic_score: float = 2.5,
         clip_skip: Optional[int] = None,
+        callback_on_step_end: Optional[Callable[[int, int, Dict], None]] = None,
+        callback_on_step_end_tensor_inputs: List[str] = ["latents", "prompt_embeds", "negative_prompt_embeds"],
```

Comment:
Should the default values for all pipelines just be set to `["latents"]`? Right now these default values vary across the pipelines, but I don't see why they should.

Reply:
Agree think by default we should only pass the `latents`

Reply:
@patrickvonplaten 
I also included `timesteps` in the default - would this be ok?  all pipelines have `timesteps` and `latents` available for callbacks (the test assume that they have it). it is a very small tensor too

Reply:
happy to update to only `latents` though!

Reply:
I actually don't think we should pass the `timesteps` at all. Users should not mess with `timesteps`. 

In a v1, let's only focus on `latents`

Reply:
Hmm so what is the advantage of having an  `callback_on_step_end_tensor_inputs` argument?  If the allowed tensors are hardcoded in `_callback_tensor_inputs`  why not always pass them into the `callback_kwargs`?

Reply:
@patrickvonplaten 
I will only leave `latents` in default for now 
But I think it is super important to make `timesteps` available as a "read-only" variable - otherwise you would not even know which step is last one etc, and one of the test I wrote won't be able to run 

agree it's user should not mess with `timesteps` so I made it read only here. e.g. below code is taken from stable diffusion pipeline. Even though user can add it to `callback_kwargs` and update them, the result won't be read back 

```python
if callback_on_step_end is not None:
                    callback_kwargs = {}
                    for k in callback_on_step_end_tensor_inputs:
                        callback_kwargs[k] = locals()[k]
                    callback_outputs = callback_on_step_end(self, i, t, callback_kwargs)

                    latents = callback_outputs.pop("latents", latents)
                    prompt_embeds = callback_outputs.pop("prompt_embeds", prompt_embeds)
                    negative_prompt_embeds = callback_outputs.pop("negative_prompt_embeds", negative_prompt_embeds)
```

Reply:
oh I think I often forget that tensors can be modified in place so there is really no way to make it "read-only" once we make it available ..... 

I removed the `"timesteps"` from the callback_tensor_inputs list in all pipelines. But I added a private variable on pipeline `_num_timesteps` for now.... Just think the callback function really need to be aware the total number of steps and that information is not available right now... we have `num_inference_steps` but that's often different from the actual number of steps for the denoising loop because of `strength` and how different scheduler works etc 

happy to reiterate if you have any feedbacks!